atproto/auth/oauth/types.go

```diff
@@ -407,3 +407,23 @@ type TokenResponse struct {
 	// Refresh token, for doing additional token requests to the auth server.
 	RefreshToken string `json:"refresh_token"`
 }
+
+// Returned by [ClientApp.ProcessCallback] if the AS signals an error in the redirect URL parameters, per rfc6749 section 4.1.2.1
+//
+// NOTE: This is untrusted data and should not be e.g. rendered to HTML without appropriate escaping
+type ErrCallback struct {
+	code        string
```

Comment:
A bit pedantic, but to distinguish "error code" from "HTTP status code", and "error URI" from "HTTP request URI" or "OAuth Redirect URI", I would name these fields:

- `ErrorCode`
- `ErrorDescription`
- `ErrorURI`